### PR TITLE
feature(provider): logging, tests and refactor

### PR DESCRIPTION
**Pull request by julien51**
# Description

Addinglogging to sentry + basic tests and refactored a bit.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: provider/src/index.ts [hunk 1]
```diff
@@ -27,120 +9,22 @@ export default {
...
-    // Reject requests that are not POST
-    if (request.method !== 'POST') {
-      return new Response(JSON.stringify({ message: 'Method not supported' }), {
-        status: 400,
-        headers: {
-          'content-type': 'application/json',
-        },
-      })
-    }
+    context.passThroughOnException()
```
Comment (julien51) [inline provider/src/index.ts:14]: This should pass the errors down to the front-end apps.

File: provider/src/supportedNetworks.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { Env } from './types'
+
+// This is the list of networks currently supported
+const supportedNetworks = (env: Env, networkId: string) => {
```
Comment (julien51) [inline provider/src/supportedNetworks.ts:4]: Moving to env vars (already configured in Cloudflare) to avoid exposing them publicly.

Comment (searchableguy) [reply provider/src/supportedNetworks.ts:4]: Wouldn't it be easier to get rid of the switch statement and simply use object then return object[id]?

**Review by searchableguy — APPROVED** [review]
LGTM

File: provider/src/supportedNetworks.ts (hunk 2 above)
Comment (julien51) [reply provider/src/supportedNetworks.ts:4]: Ha, we need to pass `env` though :/

Comment (searchableguy) [reply provider/src/supportedNetworks.ts:4]: That would be fine, right?

```ts
const object = { 
  "1": env.MAINNET, 
  "5": env.GOERLI 
}

return object[id]
```

inside the function 

It's just a little easier to read than a switch case maze.

Comment (julien51) [reply provider/src/supportedNetworks.ts:4]: Ha! Sorry i had misunderstood. Yes. That works too. I'll fix it.

**Pull request merged**